enumerables.rb

```diff
@@ -0,0 +1,75 @@
+module Enumerable
+  def my_each
+    length.times { |value| yield(self[value]) }
+    self
+  end
+
+  def my_each_with_index
+    length.times { |value| yield(self[value], value) }
+    self
+  end
+
+  def my_select
+    array = []
+    my_each { |num| array << num if yield(num) }
+    array
+  end
+
+  def my_all?
+    result = true
+    my_each { |i| break result = false unless yield(i) }
+    result
+  end
+
+  def my_any?
+    result = false
+    my_each { |i| break result = true if yield(i) }
+    result
+  end
+
+  def my_none?
+    result = true
+    my_each { |i| break result = false if yield(i) }
+    result
+  end
+
+  def my_count(arg)
+    counter = 0
+    my_each { |i| counter += 1 if arg == i }
+    counter
+  end
+
+  def my_map(proc = nil)
+    array = []
+    block_given? ? my_each { |i| array << yield(i) } : my_each { |i| array << proc.call(i) }
+    array
+  end
```

Comment:
- [ ] `my_map` is not working as the original `map` method, also is not passing all test cases.

### `map` is an Enumerable that has these characteristics:
- Returns an Enumerator if no block is given.
```
[1, 2, 3].map
# #<Enumerator: [1, 2, 3]:map>
```
- Returns a **new array** with the results of running a block
```
[1, 2, 3].map { |num| num * 2 }
# [2, 4, 6]
```
- doesn't change/modify the original array/range/hash
- can accept a proc as an argument
```
['1', '2', '3'].map(&:to_i)
# [1, 2, 3]
```
[Enumerable#map](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-map)